doc/apiv3-documentation.apib

```diff
@@ -3390,41 +3436,306 @@ For more details and all possible responses see the general specification of [Fo
+
+The response will be user added as watcher.
+In case the user was already watching the work package an `HTTP 200` is returned, an
+`HTTP 201` if the user was added as a new watcher.
+
++ Parameters
+    + work_package_id (required, integer, `1`) ... Work package id
+
++ Request (application/json)
+
+        {
+            "user": { "href": "/api/v3/users/1" }
+        }
+
 + Response 200 (application/hal+json)
 
     [User][]
 
++ Response 201 (application/hal+json)
+
+    [User][]
+
++ Response 400 (application/hal+json)
+
+    Occurs when the client did not send a valid JSON object in the request body.
+
+    For example:
+
+    * The request did not contain a single JSON object
+    * The JSON object did not contain the key `user`
+    * The value of `users` was not a link object
+
+    + Body
+
+            {
+                "_type": "Error",
+                "errorIdentifier": "urn:openproject-org:api:v3:errors:InvalidRequestBody",
+                "message": "The request body was not a single JSON object."
+            }
+
++ Response 403 (application/hal+json)
+
+    Returned if the client does not have sufficient permissions.
+
+    **Required permission:**
+
+    * view work package (for self)
+    * add work package watchers (for other users)
+
+    *Note that you will only receive this error, if you are at least allowed to see the corresponding work package.*
+
+    + Body
+
+            {
+                "_type": "Error",
+                "errorIdentifier": "urn:openproject-org:api:v3:errors:MissingPermission",
+                "message": "You are not allowed to add watchers to this work package."
+            }
+
++ Response 404 (application/hal+json)
+
+    Returned if the work package does not exist or the client does not have sufficient permissions to see it.
+
+    **Required permission:** view work package
+
+    *Note that you will effectively not be able to change the watchers of a work package without being able to see the work package.*
+
+    + Body
+
+            {
+                "_type": "Error",
+                "errorIdentifier": "urn:openproject-org:api:v3:errors:NotFound",
+                "message": "The specified work package does not exist."
+            }
+
++ Response 422 (application/hal+json)
+
+    Returned if:
+
+    * the client tries to specify a link to a resource that is not a user (`ResourceTypeMismatch`)
+    * the user specified is not allowed to watch that work package (`PropertyConstraintViolation`)
+    * the user specified does not exist (`PropertyConstraintViolation`)
+
+    + Body
+
+            {
+                "_type": "Error",
+                "errorIdentifier": "urn:openproject-org:api:v3:errors:ResourceTypeMismatch",
+                "message": "Expected resource of type 'User', but got a 'Status'.",
+            }
+
 ## Remove Watcher [/api/v3/work_packages/{work_package_id}/watchers/{id}]
```

Comment:
I guess there will be more occurences of such errors.

One (anyone) could do this in a ticketless cleanup PR IMO, I am too lazy to do it in this PR TBH.